core/src/OpenACC/Kokkos_OpenACC_Instance.cpp

```diff
@@ -94,3 +94,10 @@ int Kokkos::Experimental::OpenACC::concurrency() const {
   return Impl::OpenACCInternal::m_concurrency;
 }
 #endif
+
+void Kokkos::Experimental::Impl::create_OpenACC_instances(
+    std::vector<OpenACC>& instances) {
+  for (int s = 0; s < int(instances.size()); s++) {
+    instances[s] = OpenACC(s);
+  }
```

Comment:
This means that we have multiple instances synchronizing when calling `partition_space` multiply AFAICT. Should we rather keep track of the highest number assigned to make sure that doesn't happen?

Reply:
You are right, but the current OpenACC backend implementation does not support host-side multithreading, except for offloading parallel constructs into host as a multicore CPU target. We will have to revisit this when the OpenACC backend supports host-side multithreading.

Reply:
Added.